src/Data/Nat/Show.agda

```diff
@@ -8,29 +8,104 @@
 
 module Data.Nat.Show where
 
-open import Data.Nat
+open import Data.Char as Char using (Char)
+open import Data.Digit using (showDigit; toDigits)
+open import Function   using (_∘_; _$_)
+open import Data.List  using (List; []; _∷_; map; reverse)
+open import Data.Nat   using (ℕ; _≟_; suc; pred; _+_; _*_; _<_; _>_; z≤n; s≤s;
+                                                                        _≤?_)
+open import Data.Nat.DivMod     using (_div_; _%_; a≡a%n+[a/n]*n)
+open import Data.Product        using (proj₁)
+open import Data.Nat.Properties using
+                     (<-irrefl; m≢0⇒suc[pred[m]]≡m; +-comm; *-identityʳ;
+                                         m≤m+n; *-monoʳ-<; module ≤-Reasoning)
+open import Data.String as Str using (String)
+import      Induction.Nat
+open import Induction.WellFounded using (Acc)
+open import Relation.Binary.PropositionalEquality using (_≡_; refl; sym;
+                                                         cong)
+open import Relation.Nullary           using (yes; no)
 open import Relation.Nullary.Decidable using (True)
-open import Data.String.Base as String using (String)
-open import Data.Digit
-open import Data.Product using (proj₁)
-open import Function
-open import Data.List.Base
+open import Relation.Nullary.Negation  using (contradiction)
 
--- showInBase b n is a string containing the representation of n in
--- base b.
 
+------------------------------------------------------------------------------
+open ≤-Reasoning
+
+open Acc using (acc)
+
+<-acc :  {n : ℕ} → Acc _<_ n
+                          -- This means that _<_ is well-founded on ℕ
+                          -- ("each number in ℕ is accessible from 0 by _<_").
+<-acc {n} = Induction.Nat.<-wellFounded n
+
+
+-- Conversion from unary representation to the representation by the given
+-- base. 
+toDigitNats : (base : ℕ) → base > 1 → ℕ → List ℕ
+toDigitNats 0             ()
+toDigitNats 1             1>1    _ =  contradiction 1>1 (<-irrefl refl)
```

Comment:
This line can be simplified by pattern matching on `1>1` as `(s≤s ())`

Reply:
Done. See the second commit of April 6.